mock-api/msw/handlers.ts

```diff
@@ -1601,26 +1603,80 @@ export const handlers = makeHandlers({
     return handleOxqlMetrics(params.body)
   },
   siloMetric: handleMetrics,
+  affinityGroupList: ({ query }) => {
+    const project = lookup.project({ ...query })
+    const affinityGroups = db.affinityGroups.filter((i) => i.project_id === project.id)
+    return paginated(query, affinityGroups)
+  },
+  affinityGroupView: ({ path, query }) => lookup.affinityGroup({ ...path, ...query }),
+  affinityGroupMemberList: ({ path, query }) => {
+    const project = lookup.project({ ...query })
+    const affinityGroup = lookup.affinityGroup({
+      project: project.id,
+      affinityGroup: path.affinityGroup,
+    })
+    const members: Json<AffinityGroupMember>[] = db.affinityGroupMemberLists
+      .filter((i) => i.affinity_group_id === affinityGroup.id)
+      .map((i) => {
+        const { id, name, run_state } = lookup.instance({
+          instance: i.affinity_group_member.value.id,
+        })
+        return { type: 'instance', value: { id, name, run_state } }
+      })
+    return { items: members }
+  },
+  antiAffinityGroupList: ({ query }) => {
+    const project = lookup.project({ ...query })
+    const antiAffinityGroups = db.antiAffinityGroups.filter(
+      (i) => i.project_id === project.id
+    )
+    return paginated(query, antiAffinityGroups)
+  },
+  antiAffinityGroupView: ({ path, query }) =>
+    lookup.antiAffinityGroup({ ...path, ...query }),
+  antiAffinityGroupMemberList: ({ path, query }) => {
+    const project = lookup.project({ ...query })
+    const antiAffinityGroup = lookup.antiAffinityGroup({
+      project: project.id,
+      antiAffinityGroup: path.antiAffinityGroup,
+    })
```

Comment:
The group lookup handles the project lookup internally:

```suggestion
    const antiAffinityGroup = lookup.antiAffinityGroup({ ...path, ...query })
```